src/main/java/edu/umich/ctools/esb/utils/WAPI.java

```diff
@@ -80,9 +81,23 @@
 	private String token;
 	private String grant_type_value = CLIENT_CREDENTIALS;
 	private String scope_value = PRODUCTION;
-	
+
 	private String renewal;
-	
+
+	// number of requests to the web service.
+	private int requestCount = 0;
+
+	/********************/
+	// Allow periodic forced token renewals.  Particularly useful for checking token renewal behavior.
+
+	// keep track of the number of renewals.
+	private int tokenRenewalCount = 0;
+	// set count for how often should force renewal. 0 implies
+	// never force renewal.
+	private int tokenRenewalFrequency = 0;
```

Comment:
can the name be `tokenForcedRenewalFrequency`?

Reply:
Good idea.  I'll do it.